pinot-spi/src/main/java/org/apache/pinot/spi/plugin/PluginClassLoader.java

```diff
@@ -36,18 +36,9 @@
   public PluginClassLoader(URL[] urls, ClassLoader parent) {
     super(urls, parent);
     classLoader = PluginClassLoader.class.getClassLoader();
-    Method method = null;
-    try {
-      method = URLClassLoader.class.getDeclaredMethod("addURL", URL.class);
-
-    } catch (NoSuchMethodException e) {
-      //this should never happen
-      ExceptionUtils.rethrow(e);
-    }
-    method.setAccessible(true);
     for (URL url : urls) {
       try {
-        method.invoke(classLoader, url);
+        super.add(url);
```

Comment:
what's the behavior for java8 on this?

Reply:
I think we need to check on jdk version and choose different APIs here.

Reply:
Will do!

Reply:
Update: In java9+ the application class loader no longer extends URIClassLoader: https://community.oracle.com/tech/developers/discussion/4011800/base-classloader-no-longer-from-urlclassloader

Invoking addURL will work for all jdk versions. In the findClass method, if the class is not in the parent class loader it will be in this one. I added a comment to explain.

Also, this has been working for us in production.